src/components/_modal/pages/material/Material.page.js

```diff
@@ -0,0 +1,227 @@
+    loanDateAnalysis(dueDateString);
+
+  return (
+    <>
+      <Text type="text2" className={styles.spacer}>
+        {Translate({ context: "profile", label: "to-return" })} {dateString}
+      </Text>
+      <div className={styles.status}>
+        <Icon
+          className={styles.ornament}
+          size={{ w: 5, h: "auto" }}
+          src={"ornament1.svg"}
+          alt=""
+        />
+        {isOverdue ? (
+          <Text type="text2">
+            {Translate({
+              context: "profile",
+              label: "date-overdue",
+            })}
+          </Text>
+        ) : (
+          <Text
+            type="text2"
+            className={cx({ [styles.isWarning]: isCountdown })}
+          >
+            {daysToDueDateString}
+          </Text>
+        )}
+      </div>
+    </>
+  );
+};
+
+const DynamicColumnOrder = ({ pickUpExpiryDate, holdQueuePosition }) => {
+  const pickUpDate = new Date(pickUpExpiryDate);
+  const isReadyToPickup = !!pickUpExpiryDate;
+  const dateString = isReadyToPickup ? dateToDayInMonth(pickUpDate) : null;
+
+  if (isReadyToPickup) {
+    return (
+      <>
+        <Text type="text1" tag="span" className={styles.isReady}>
+          {Translate({
+            context: "profile",
+            label: "ready-to-pickup",
+          })}
+        </Text>
+        <div className={styles.status}>
+          <Icon
+            className={styles.ornament}
+            size={{ w: 5, h: "auto" }}
+            src={"ornament1.svg"}
+            alt=""
+          />
+          <Text type="text2" tag="span">
+            {Translate({
+              context: "profile",
+              label: "pickup-deadline",
+            })}
+             {dateString}
+          </Text>
+        </div>
+      </>
+    );
+  }
+
+  return (
+    <div className={styles.status}>
+      <Icon
+        className={styles.ornament}
+        size={{ w: 5, h: "auto" }}
+        src={"ornament1.svg"}
+        alt=""
+      />
+      <Text type="text2" tag="span">
+        {holdQueuePosition === "1"
+          ? `${Translate({
+              context: "profile",
+              label: "front-of-row",
+            })}`
+          : `${holdQueuePosition - 1} ${Translate({
+              context: "profile",
+              label: "in-row",
+            })}`}
+      </Text>
+    </div>
+  );
+};
+
+const Material = ({ context }) => {
+  const {
+    label,
+    title,
+    creator,
+    materialType,
+    creationYear,
+    image,
+    workId,
+    dueDateString,
+    type,
+    pickUpExpiryDate,
+    holdQueuePosition,
+  } = context;
+
+  const renderDynamicContent = () => {
+    switch (type) {
+      case "LOAN":
+        return <DynamicContentLoan dueDateString={dueDateString} />;
+      case "ORDER":
+        return (
+          <DynamicColumnOrder
+            pickUpExpiryDate={pickUpExpiryDate}
+            holdQueuePosition={holdQueuePosition}
+          />
+        );
+    }
+  };
```

Comment:
2 ting:
1. Mangler den ikke en default? (samme for `renderDynamicButton`)

2. (Kan ignoreres, men jeg synes det ser lidt underligt ud her, særligt fordi jeg personligt foretrækker at komponenter er komponenter og ikke funktioner) 
- Hvis du giver dem parametre med i stedet for at scope dem som nu, så kan du lægge dem i global kontekst i stedet. (Samme princip for `renderDynamicButton` --> `RenderDynamicButton`)

```
  const RenderDynamicContent = (...parameterObject) => {
    switch (type) {
      case "LOAN":
        return <DynamicContentLoan dueDateString={parameterObject?.dueDateString} />;
      case "ORDER":
        return (
          <DynamicColumnOrder
            pickUpExpiryDate={parameterObject?.pickUpExpiryDate}
            holdQueuePosition={parameterObject?.holdQueuePosition}
          />
        );
    }
  };
```

og i Material-komponenten:
```
const parameterObjectForContent = {
  type: type,
  dueDateString: dueDateString,
  pickUpExpiryDate: pickUpExpiryDate,
  holdQueuePosition: holdQueuePosition,
}

...
<article ... >
...
  <div>
  ...
  <RenderDynamicContent parameterObject={parameterObjectForContent} />
  <RenderDynamicButton parameterObject={parameterObjectForButton} />
  </div>
...
</article>
```

Reply:
hmm, komponenter er funktioner. Dette er for at undgå at lave en switch i JSX, giver mere overblik i return. 
Et nyt komponent som i sig selv bare er en condition er lidt overdone i min mening

Reply:
Personligt er jeg mere til en condition i JSX'en end et nyt komponent, bare for en switch case. Hvis det er bedre end en render function?